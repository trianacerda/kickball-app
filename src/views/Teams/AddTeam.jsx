import { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { createTeam } from '../../services/teams';

function AddTeam() {
  const [name, setName] = useState('');
  const [city, setCity] = useState('');
  const [state, setState] = useState('');
  const history = useHistory();

  const onSubmit = async (e) => {
    e.preventDefault();
    const response = await createTeam({ name, city, state });
    history.push(`/teams/${response[0].id}`);
  };

  return (
    <fieldset>
      <h1>Add a Team</h1>
      <form>
        <label htmlFor="name">Name:</label>
        <input
          type="text"
          name="name"
          id="name"
          value={name}
          onChange={(e) => {
            setName(e.target.value);
          }}
        />

        <label htmlFor="city">City:</label>
        <input
          type="text"
          name="city"
          id="city"
          value={city}
          onChange={(e) => {
            setCity(e.target.value);
          }}
        />

        <label htmlFor="state">State:</label>
        <input
          type="text"
          name="state"
          id="state"
          value={state}
          onChange={(e) => {
            setState(e.target.value);
          }}
        />

        <button type="submit">Add a Team</button>
      </form>
    </fieldset>
  );
}

export default AddTeam;
