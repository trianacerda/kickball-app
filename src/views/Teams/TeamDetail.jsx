import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { getTeamById } from '../../services/teams';

function TeamDetail({ label, match }) {
  const { teamId } = match.params;
  const [team, setTeam] = useState(null);
  const [loading, setLoading] = useState(true);

  return (
    <>
      <h1>{label}</h1>
      <p>
        <Link to="/teams" className="team-link">
          Back to Team Page
        </Link>
      </p>
      <h1>{team.name}</h1>
      <p>
        From {team.city}, {team.state}
      </p>
      <ul>
        {team.players.map((player) => {
          return (
            <li key={player.id}>
              {player.position} - {player.name}
            </li>
          );
        })}
      </ul>
    </>
  );
}

export default TeamDetail;
