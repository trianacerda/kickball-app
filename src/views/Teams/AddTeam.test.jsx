import { render, screen } from '@testing-library/react';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import { MemoryRouter } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import userEvent from '@testing-library/user-event';
import AddTeam from './AddTeam';
import TeamDetail from './TeamDetail';

const mockTeam = {
  id: 9,
  created_at: '2021-12-11T02:56:49.810681+00:00',
  name: 'The couch sitters',
  city: 'sisters',
  state: 'Oregon',
  players: [],
};

const server = setupServer(
  rest.get('https://eyxqfyzkrrtcxhvkkviw.supabase.co/rest/v1/teams', (req, res, ctx) => {
    return res(ctx.json(mockTeam));
  }),
  rest.post('https://eyxqfyzkrrtcxhvkkviw.supabase.co/rest/v1/teams', (req, res, ctx) => {
    return res(ctx.json([mockTeam]));
  })
);

beforeAll(() => {
  server.listen();
});

afterAll(() => {
  server.close();
});

it('should render Form component and redirect to details apart after submit', async () => {
  const history = createMemoryHistory();
  history.push('/teams/new');

  const { container } = render(
    <MemoryRouter>
      <AddTeam />
      <TeamDetail match={{ params: { teamId: 9 } }} />
    </MemoryRouter>
  );

  screen.getByText('Add a New Team!');

  const nameField = screen.getByLabelText(/Name:/i);
  const cityField = screen.getByLabelText(/City:/i);
  const stateField = screen.getByLabelText(/State:/i);
  const submitButton = screen.getByRole('button', { name: 'Submit' });

  userEvent.type(nameField, 'The couch sitters');
  userEvent.type(cityField, 'sisters');
  userEvent.type(stateField, 'Oregon');
  userEvent.click(submitButton);

  expect(container).toMatchSnapshot();
  await screen.findByText(/sisters/i);
});
