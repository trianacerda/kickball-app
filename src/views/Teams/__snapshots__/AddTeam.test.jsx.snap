// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Form component and redirect to details apart after submit 1`] = `
<div>
  <h1>
    Add a New Team!
  </h1>
  <a
    class="teams-link"
    href="/teams"
    style="display: flex; flex-direction: start;"
  >
    Teams
  </a>
  <a
    class="player-link"
    href="/players"
    style="display: flex; flex-direction: start;"
  >
    Players
  </a>
  <fieldset>
    <form>
      <label
        for="name"
      >
        Name:
      </label>
      <input
        id="name"
        name="name"
        type="text"
        value="The couch sitters"
      />
      <label
        for="city"
      >
        City:
      </label>
      <input
        id="city"
        name="city"
        type="text"
        value="sisters"
      />
      <label
        for="state"
      >
        State:
      </label>
      <input
        id="state"
        name="state"
        type="text"
        value="Oregon"
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </fieldset>
  <h1>
    Please wait, the team is loading...
  </h1>
</div>
`;
